pkg/reconciler/common/extensions.go

```diff
@@ -22,12 +22,36 @@ import (
 	"knative.dev/operator/pkg/apis/operator/v1alpha1"
 )
 
+// NilException is a convenient null object
+var NilExtension = nilExtension{}
+
+// Extension enables platform-specific features
 type Extension interface {
-	Transformers(v1alpha1.KComponent) ([]mf.Transformer, error)
```

Comment:
It seemed redundant since a `Transformer` already returns an error. If necessary, you could return a closure around any API call to surface the error when the transform is invoked. Besides, I like inlining this call. :)

We always run the transforms immediately after creating them anyway. If that ever changes, we can revisit.

Reply:
Alrighty, I don't feel too strongly about it.